qualtran/bloqs/factoring/mod_mul.py

```diff
@@ -104,6 +116,94 @@ def wire_symbol(self, soq: 'Soquet') -> 'WireSymbol':
             return directional_text_box(f'*={self.k}', side=soq.reg.side)
 
 
+@frozen
+class MontgomeryModDbl(Bloq):
+    r"""An n-bit modular doubling gate.
+
+    This gate is designed to operate on integers in the Montgomery form.
+    Implements |x> => |2 * x % p> using $2n$ Toffoli gates.
+
+    Args:
+        bitsize: Number of bits used to represent each integer.
+        p: The modulus for the doubling.
+
+    Registers:
+        x: A bitsize-sized input register (register x above).
+
+    References:
+        [How to compute a 256-bit elliptic curve private key with only 50 million Toffoli gates](https://arxiv.org/abs/2306.08585) Fig 6d and 8
```

Comment:
can you put Fig 6d and 8 on a new line. We keep the link on the same line (even though it is >100 char) so that it shows up as a link when rendered as markdown.